flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/GeneratedAggregations.scala

```diff
@@ -36,6 +36,15 @@ abstract class GeneratedAggregations extends Function {
   def setAggregationResults(accumulators: Row, output: Row)
 
   /**
+    * Calculates the results from accumulators, and set the results to the output (with key offset)
+    *
+    * @param accumulators the accumulators (saved in a row) which contains the current
+    *                     aggregated results
+    * @param output       output results collected in a row
+    */
+  def setAggregationResultsWithKeyOffset(accumulators: Row, output: Row)
```

Comment:
We could reuse all your code, but just put it into a different method of the `CodeGenerator` and make it implement the existing methods. Their interfaces are the same.